app/decorators/prosecution_case_decorator.rb

```diff
@@ -23,4 +21,31 @@ def hearings_with_day_by_datetime
   def cracked?
     hearings.map { |h| h.cracked_ineffective_trial&.cracked? }.any?
   end
+
+  private
+
+  def sorted_hearings
```

Comment:
Have been reading about refactoring and this method is starting to smell of needing an "extract class" using inheritance or composition scenario.

[very roughly](https://refactoring.guru/extract-subclass) but if you have martin fowlers "Refactoring" or Sandi Metz's "99 Bottles of oop" they have example and explanations

Reply:
@jsugarman I have attempted to create an extract class for this method by creating a TableSorter module containing a HearingsSorter class, with subclasses HearingsProviderSorter and HearingsTypeSorter. Is this the kind of thing you had in mind and if so is the decorators folder the right location (or should they be under /lib for example)?